### PR TITLE
Create flag to control tcp routes

### PR DESCRIPTION
**Pull request by JaredGordon**
Hello --

Here's a PR to add functionality that let's users turn tcp routing on and off (along with associated load balancer, security group, listeners, route53 entries).

This is useful in environments that disallow tcp access to pas applications for security reasons.

Thanks,

-- Jared

### CODE REVIEW

File: modules/pas/lbs.tf [hunk 1]
```diff
@@ -138,10 +138,11 @@ resource "aws_lb_target_group" "ssh" {
 # TCP Load Balancer
 
 locals {
-  tcp_port_count = 10
+  tcp_port_count = "${var.use_tcp_routes ? 10 : 0}"
 }
 
 resource "aws_security_group" "tcp_lb" {
+  count       = "${var.use_tcp_routes}"
```
Comment (Polar-Beard) [inline modules/pas/lbs.tf]: Looks like this is throwing an error: 
```
Error: Error refreshing state: 2 error(s) occurred:                                                                                                                                     │ 23
                                                                                                                                                                                        │ 24 [[constraint]]
* module.pas.aws_security_group.tcp_lb: cannot parse "false" as an integer                                                                                                              │ 25   name = "github.com/onsi/ginkgo"
* module.pas.aws_lb.tcp: cannot parse "false" as an integer
```

File: modules/pas/lbs.tf [hunk 2]
```diff
@@ -164,6 +165,7 @@ resource "aws_security_group" "tcp_lb" {
 }
 
 resource "aws_lb" "tcp" {
+  count                            = "${var.use_tcp_routes}"
```
Comment (Polar-Beard) [inline modules/pas/lbs.tf]: Looks like this is throwing an error:

```
Error: Error refreshing state: 2 error(s) occurred:                                                                                                                                     │ 23
                                                                                                                                                                                        │ 24 [[constraint]]
* module.pas.aws_security_group.tcp_lb: cannot parse "false" as an integer                                                                                                              │ 25   name = "github.com/onsi/ginkgo"
* module.pas.aws_lb.tcp: cannot parse "false" as an integer
```

**Comment by Polar-Beard** [comment]
@JaredGordon Thanks for the PR 💃 

We ran into an issue running `terraform apply`. Would you mind updating with the fixes? Thanks again.

**Comment by JaredGordon** [comment]
Hi Polar-Beard: just ran through an init/plan/apply/mess with the config/re-apply on my fork and I didn't  hit this error.

What version of tf are you using? I was able to run this with v0.11.13

**Comment by notrepo05** [comment]
Hi, @JaredGordon ! We reran it with terraform v0.11.13 and encountered the same error:

We ran
```
terraform init
terraform apply
``` with `TF_VAR_use_tcp_routes` unset without encountering an error.

However after setting `TF_VAR_use_tcp_routes=false` and rerunning `terraform apply` we
received the following:
```
* module.pas.aws_security_group.tcp_lb: cannot parse "false" as an integer
* module.pas.aws_lb.tcp: cannot parse "false" as an integer
```

**Comment by notrepo05** [comment]
Hi! We ended up merging your PR in with a fix, so I'm going to close the PR. Thanks again for the PR and please ping either myself or @Polar-Beard if there are any issues. 👯‍♂️

**Comment by JaredGordon** [comment]
Thanks Nick, I've been trying to recreate the bug on my side, but can't
(tried following the steps you outlined too, no luck).

What did you end up fixing? The reason I ask (besides wanting to know what
was wrong with the PR) is that the next PR also uses these boolean flags,
and I want to avoid causing havoc again.

-- Jared

**Comment by notrepo05** [comment]
@JaredGordon It looks like our version of terraform (0.11.13) was complaining (due to its weird type system) that `count` couldn't be assigned to a boolean `"${var.use_tcp_routes}"`. [So we just turned it into an int with a ternary `count       = "${var.use_tcp_routes ? 1 : 0}"`](https://github.com/pivotal-cf/terraforming-aws/commit/ea51e725c2e87b373c75c5c92aeeb024b2e9b92d). 

It seems like it worked fine, but we didn't go as far as deploying PAS and testing all the functionality; we're having issues with our `terraforming-aws` pipeline that automates that test. Once it's fixed we'll be able to more easily confirm it's working properly. That being said, the `apply` succeeded. So I don't expect future PRs to break due to this this one.

We're also on the Mac OSX version of terraform, so maybe that accounts for not being able to reproduce our issue?

**Comment by JaredGordon** [comment]
it's a mystery: ran these on my mac too...

Anyway, thanks for your help. I've pulled together and submitted the next
PR (https://github.com/pivotal-cf/terraforming-aws/pull/126) which is
simpler than the TCP one. It makes use of the booleans too, but I've gone
ahead and used the tertiary operator to hopefully sidestep the issues with
the TCP PR.

Have a good weekend!

**Pull request closed without merging**